std/conv.d

```diff
@@ -5334,3 +5334,249 @@ unittest
         auto y = castFrom!(long*).to!int(x);
     }
 }
+
+/**
+The function $(D isHexLiteral) checks the consistency of a string for the
+functions $(D hexString) and $(D hexBytes).
+The result is only true if the input string is composed of ASCII white
+characters (\f\n\r\t\v) or hexadecimal digits (regardless of the case).
+
+It can be used before calling the run time version of $(D hexString) and
+$(D hexBytes) since they may throw an assert error if the input string is
+invalid.
+*/
+static bool isHexLiteral(StringType)(in StringType hexData)
+{
+    import std.ascii;
+    size_t i;
+    bool isH;
+    foreach(c; hexData)
+    {
+        isH = c.isHexDigit;
+        if ((!isH) & (!c.isWhite))
+            return false;
+        i += isH;
+    }
+    return !(i & 1) & (i > 0);
+}
```

Comment:
Use logical operators for logical operations and bit operators for bit operations.